From: QbProg <tholag@gmail.com>
Date: Sat, 11 Jun 2011 15:08:13 +0200
Subject: [PATCH 25/82] [warning-fix][deprecated-conversion-from-string]

Fixed warning: deprecated conversion from string constant to 'char*'
---
 src/Aspect/Aspect_Driver.cxx           |    2 +-
 src/DPrsStd/DPrsStd.cxx                |    2 +-
 src/Graphic3d/Graphic3d_Texture1D.cxx  |    2 +-
 src/Graphic3d/Graphic3d_Texture2D.cxx  |    2 +-
 src/Graphic3d/Graphic3d_TextureEnv.cxx |    2 +-
 src/OpenGl/OpenGl_degmodel.cxx         |    2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

From 1b35e5df96313a8e2f70d8f44c37287f9d75dd84 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Wed, 20 Jul 2011 22:50:16 +0000
Subject: [PATCH 15/26] [warning-fix][const-char*]
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Replace 'char*' by 'const char*'

gcc 4.3 print this warning:
  deprecated conversion from string constant to ‘char*’

Note: this commit breaks API compatibility
---
 inc/OpenGl_Extension.hxx                       |    4 ++--
 inc/OpenGl_TextRender.hxx                      |    8 ++++----
 inc/OpenGl_tgl_util.hxx                        |    2 +-
 src/OpenGl/OpenGl_Extension.cxx                |    4 ++--
 src/OpenGl/OpenGl_tgl_util.cxx                 |    2 +-
 src/OpenGl/OpenGl_togl_begin_immediat_mode.cxx |    2 +-
 src/OpenGl/OpenGl_txgl.cxx                     |    4 ++--
 7 files changed, 13 insertions(+), 13 deletions(-)

From 140dc622232d82d5c65849f8ab0a56ac655b2890 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 22 Jul 2011 12:34:02 +0200
Subject: [PATCH 22/26] [warning-fix][const-char*]

Oops, forgot this chunk in 1b35e5d
---
 inc/OpenGl_Extension.hxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From 58f1e9f293d52823262a1ef022eab5d3f8b63ac5 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Fri, 22 Jul 2011 18:14:23 +0200
Subject: [PATCH 23/26] [warning-fix][const-char*]

Updated char* to const char* in OpenGl_Extension.cxx
---
 src/OpenGl/OpenGl_Extension.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From 7c7881e84491cb45dede027ebcef03137d84e72a Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 22 Jul 2011 20:05:12 +0000
Subject: [PATCH 24/26] [warning-fix][const-char*]

Remove const_cast from QueryExtensionGLX calls, this
is no more needed; but OpenGl_QueryExtension and
OpenGl_QueryExtensionGLX are unused and may get
dropped instead.
---
 src/OpenGl/OpenGl_GraphicDriver.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

From 615ca745ba0c2933b4954367054e35759404c2a8 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Mon, 25 Jul 2011 14:13:27 +0200
Subject: [PATCH 2/4] Let first argument of GetEnv() be a const char*, like
 with getenv() on Unix

---
 inc/InterfaceGraphic_wntio.hxx                    |    2 +-
 src/InterfaceGraphic/InterfaceGraphic_cPrintf.cxx |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

From ea1698b5caad03a3f5e4bb0f8aba4a0cd2bc3633 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Mon, 25 Jul 2011 14:17:07 +0200
Subject: [PATCH 3/4] In printf-like functions, fmt argument can be a const

---
 inc/InterfaceGraphic_wntio.hxx                    |    4 ++--
 src/InterfaceGraphic/InterfaceGraphic_cPrintf.cxx |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

From 3487f37db04a2c1497bd3bdcbd5f139eb6471fb6 Mon Sep 17 00:00:00 2001
From: Hugues Delorme <delorme.hugues@fougsys.fr>
Date: Tue, 26 Jul 2011 16:34:03 +0200
Subject: [PATCH 2/3] [warning-fix][deprecated-conversion-from-string]

Fix proposed by Denis Barbier : modify global variable
'console_command' instead of parameter 'filename'.
---
 src/Draw/Draw.cxx                         |    6 +++---
 src/Graphic3d/Graphic3d_GraphicDevice.cxx |    7 ++++---
 2 files changed, 7 insertions(+), 6 deletions(-)

Index: oce/inc/InterfaceGraphic_wntio.hxx
===================================================================
--- oce.orig/inc/InterfaceGraphic_wntio.hxx
+++ oce/inc/InterfaceGraphic_wntio.hxx
@@ -31,9 +31,9 @@ extern "C" {
 #  endif  /* __InterfaceGraphic_DLL */
 # endif  /* __InterfaceGraphic_API */
 
-__InterfaceGraphic_API int   cPrintf ( char*, ... );
-__InterfaceGraphic_API int   fcPrintf ( int, char*, ... );
-__InterfaceGraphic_API char* GetEnv ( char* );
+__InterfaceGraphic_API int   cPrintf ( const char*, ... );
+__InterfaceGraphic_API int   fcPrintf ( int, const char*, ... );
+__InterfaceGraphic_API char* GetEnv ( const char* );
 
 # ifdef __cplusplus
 }
Index: oce/src/Aspect/Aspect_Driver.cxx
===================================================================
--- oce.orig/src/Aspect/Aspect_Driver.cxx
+++ oce/src/Aspect/Aspect_Driver.cxx
@@ -49,7 +49,7 @@ Aspect_Driver::Aspect_Driver () {
 
   OSD_Environment CSF_MDTVFontDirectory ;
 
-  char *fontdir , *casroot ;
+  const char *fontdir , *casroot ;
   fontdir =  getenv("CSF_MDTVFontDirectory" );
   if ( !fontdir) {
     casroot  = getenv("CASROOT");
Index: oce/src/DPrsStd/DPrsStd.cxx
===================================================================
--- oce.orig/src/DPrsStd/DPrsStd.cxx
+++ oce/src/DPrsStd/DPrsStd.cxx
@@ -69,7 +69,7 @@ static Standard_Boolean FoundCAFPlugin (
 
  // Define Environment Variable 
 
-  char *plugin, *casroot, *standard;
+  const char *plugin, *casroot, *standard;
   plugin   = getenv("CSF_PluginDefaults");
   standard = getenv("CSF_StandardDefaults");
   casroot  = getenv("CASROOT");
Index: oce/src/Graphic3d/Graphic3d_Texture1D.cxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d_Texture1D.cxx
+++ oce/src/Graphic3d/Graphic3d_Texture1D.cxx
@@ -42,7 +42,7 @@ static TCollection_AsciiString GetEnvir
   static Standard_Boolean IsDefined=Standard_False ;
   static TCollection_AsciiString VarName;
   if ( !IsDefined ) {
-    char *envir, *casroot ;
+    const char *envir, *casroot ;
     envir = getenv("CSF_MDTVTexturesDirectory") ;
     
     Standard_Boolean HasDefinition = Standard_False ;
Index: oce/src/Graphic3d/Graphic3d_Texture2D.cxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d_Texture2D.cxx
+++ oce/src/Graphic3d/Graphic3d_Texture2D.cxx
@@ -38,7 +38,7 @@ static TCollection_AsciiString GetEnvir
   static Standard_Boolean IsDefined=Standard_False ;
   static TCollection_AsciiString VarName;
   if ( !IsDefined ) {
-    char *envir, *casroot ;
+    char const*envir, *casroot ;
     envir = getenv("CSF_MDTVTexturesDirectory") ;
     
     Standard_Boolean HasDefinition = Standard_False ;
Index: oce/src/Graphic3d/Graphic3d_TextureEnv.cxx
===================================================================
--- oce.orig/src/Graphic3d/Graphic3d_TextureEnv.cxx
+++ oce/src/Graphic3d/Graphic3d_TextureEnv.cxx
@@ -39,7 +39,7 @@ static TCollection_AsciiString GetEnvir
   static Standard_Boolean IsDefined=Standard_False ;
   static TCollection_AsciiString VarName;
   if ( !IsDefined ) {
-    char *envir, *casroot ;
+    char const *envir, *casroot ;
     envir = getenv("CSF_MDTVTexturesDirectory") ;
     
     Standard_Boolean HasDefinition = Standard_False ;
Index: oce/src/InterfaceGraphic/InterfaceGraphic_cPrintf.cxx
===================================================================
--- oce.orig/src/InterfaceGraphic/InterfaceGraphic_cPrintf.cxx
+++ oce/src/InterfaceGraphic/InterfaceGraphic_cPrintf.cxx
@@ -53,7 +53,7 @@ public:
 
 static Init_ init;
 
-int cPrintf ( char* fmt, ... ) {
+int cPrintf ( const char* fmt, ... ) {
 
   static BOOL   first = TRUE;
   static HANDLE hConsole = NULL;
@@ -82,7 +82,7 @@ int cPrintf ( char* fmt, ... ) {
 
 }  /* end cPrintf */
 
-int fcPrintf ( int dummy, char* fmt, ... ) {
+int fcPrintf ( int dummy, const char* fmt, ... ) {
 
   static BOOL   first = TRUE;
   static HANDLE hConsole = NULL;
@@ -111,7 +111,7 @@ int fcPrintf ( int dummy, char* fmt, ...
 
 }  /* end fcPrintf */
 
-char* GetEnv ( char* name ) {
+char* GetEnv ( const char* name ) {
 
   DWORD dwLen;
   PENV  env = ( PENV )TlsGetValue ( tlsIndex );
